tests/unit/multieditor/test_actions.py

```diff
@@ -0,0 +1,911 @@
+        }
+    }
+    expected_map = {
+        'key_a': {
+            'key_b': 'success',
+            'key_c': 'test'
+        }
+    }
+    custom_schema = {
+        "properties": {
+            "key_a": {
+                "properties": {
+                    "key_b": {
+                        "type": "string"
+                    },
+                    "key_c": {
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "value"
+                ],
+                "type": "object"
+            },
+            "type": "object",
+        },
+        "type": "object",
+    }
+    add = Addition(keypath=['key_a', 'key_b'], value="success")
+    add.apply(record, custom_schema)
+    assert record == expected_map
+
+
+def test_addition_array_with_condition(get_schema):
+    """should test record addition for object using condition check"""
+    record = {
+        'key_a': {
+            'key_b': ['Hello'],
+            'key_c': 'test'
+        }
+    }
+
+    expected_map = {
+        'key_a': {
+            'key_b': ['Hello', 'World'],
+            'key_c': 'test'
+        }
+    }
+
+    custom_schema = {
+        "properties": {
+            "key_a": {
+                "properties": {
+                    "key_b": {
+                        "items": {
+                            "type": "string"
+                        },
+                        "type": "array",
+                    },
+                    "key_c": {
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "value"
+                ],
+                "type": "object"
+            },
+            "type": "object",
+        },
+        "type": "object",
+    }
+    add = Addition(keypath=['key_a', 'key_b'],
+                   conditions=[{'keypath': ['key_a', 'key_c'],
+                                'match_type': 'exact',
+                                'value':'test'}],
+                   value="World")
+    add.apply(record, custom_schema)
+    assert record == expected_map
+
+
+def test_addition_array(get_schema):
+    """should test record addition for nested array"""
+    record = {
+        "titles": [
+            {
+                "title": "test"
+            },
+            {
+                "title": "test"
+            }
+        ],
+        "document_type": ["book"]
+    }
+    expected_map = {
+        "titles": [
+            {
+                "title": "test",
+                "subtitle": "success"
+            },
+            {
+                "title": "test",
+                "subtitle": "success"
+            }
+        ],
+        "document_type": ["book"]
+    }
+    add = Addition(keypath=['titles', 'subtitle'], value="success")
+    add.apply(record, get_schema)
+    assert record == expected_map
+
+
+def test_addition_array_with_condition_condition(get_schema):
```

Comment:
What is `condition_condition`? can you clarify in the name of the test? Don't be shy with test names, it's better to make them understandable than short named.